lib/Compiler.js

```diff
@@ -340,6 +341,7 @@ class Compiler extends Tapable {
 						source.existsAt = targetPath;
 						source.emitted = true;
 						this.outputFileSystem.writeFile(targetPath, content, callback);
+						source._cachedSource = undefined;
```

Comment:
`writeFile` is async and potentially may fail; does it change how `_cachedSource` should be cleared? (for ex., only on successful write, or only after the async operation is complete)

Reply:
If `writeFile` fails, wouldn't the whole process fail?

Reply:
That's an assumption this piece of code should not make.

Reply:
Also this code currently seems to assume that `writeFile` uses `_cachedSource` synchronously which may not be the case depending on `writeFile` implementation.

Reply:
Say a single file's cache wasn't cleared, it would have the memory footprint of a single file (not a big deal).

Say a single file's cache was prematurely pruged, webpack knows how to repopulate the cache so it wouldn't lose the source (it's only a cache)

Reply:
Otherwise, if `writeFile` does not in fact rely on `_cachedSource`, there has to be a reason to put this clearing below the `writeFile` call and not above where other changes to `source` state are made.

Reply:
Repopulating the cache may also be expensive. There has to be a clear lifecycle for the cache, otherwise it looks like blind troubleshooting that works by coincidence, not by design. I'm not saying this solution is wrong, but it's not clear why it's right either.

Reply:
What makes you think that `writeFile` could be asynchronous? The forEachLimit is the async part

Reply:
What should be clear is that the cache is never purged and piles up to a GB+. Removing it right after it's written is the obvious end of life for a cache element.

Reply:
> What makes you think that writeFile could be asynchronous? The forEachLimit is the async part

The fact that a `callback` is passed to it. Is the `writeFile` call guaranteed to finish all its work synchronously despite receiving a callback?

> What should be clear is that the cache is never purged and piles up to a GB+. Removing it right after it's written is the obvious end of life for a cache element.

Yes, I get that. That's why I'm trying to clarify when this "after it's written" event happens. Is it guaranteed to happen synchronously within the `writeFile` call? If so, the code is correct.

Reply:
It's getting pretty late here (2am) I'll take another look at what you're saying in the morning :+1:

Reply:
> It's getting pretty late here (2am) I'll take another look at what you're saying in the morning 👍

Yes, it's 12am on my end. Thanks for addressing the memory issue and paying attention to my comments!

Reply:
@sompylasar I looked into it and it makes sense to purge the cache before `writeFile` and I tested it and the fix still works. Thanks for your feedback :+1:

Reply:
@sompylasar :thinking: What do you make of the failing `codecov/changes/integration`?